src/PackageSelection/PackageSelection.php

```diff
@@ -711,32 +707,17 @@ private function selectLinks(Pool $pool, array $links, bool $isRoot, bool $verbo
             } elseif (is_a($link, Link::class)) {
                 $name = $link->getTarget();
                 if (!$isRoot && $this->onlyBestCandidates) {
-                    $selector = new VersionSelector($pool);
+                    $selector = new VersionSelector($repositorySet);
                     $matches = [$selector->findBestCandidate($name, $link->getConstraint()->getPrettyString())];
                 } else {
-                    $matches = $pool->whatProvides($name, $link->getConstraint(), true);
+                    $matches = $repositorySet->createPoolForPackage($link->getTarget())->whatProvides($name, $link->getConstraint());
                 }
 
                 if (0 === \count($matches)) {
                     $this->output->writeln('<error>The ' . $name . ' ' . $link->getPrettyConstraint() . ' requirement did not match any package</error>');
                 }
             }
 
-            if (!$isRoot && $this->requireDependencyFilter && \count($matches) > 1) {
```

Comment:
I removed this block in 6330be222cc3641e0f47cadd8af2609ec27500fb.

Really not understood what It was supposed to do. The packages does not have an `id` when they comes from the `RepositorySet` (line 710), not the `Pool` (line 713).